core/src/test/java/ai/timefold/solver/core/testutil/AbstractMeterTest.java

```diff
@@ -16,7 +18,10 @@
 import io.micrometer.core.instrument.simple.SimpleConfig;
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
 
-@ResourceLock(value = "metrics") // Mutual exclusion to prevent flaky tests.
+// Mutual exclusion to prevent flaky tests from different test suites
+@ResourceLock(value = "metrics")
+// We also want to ensure that tests of the same suite do not update the registry concurrently
+@Execution(ExecutionMode.SAME_THREAD)
```

Comment:
Maybe we only want the latter annotation?
It seems that the resource lock did not work.